tests/manage/pv_services/test_rwo_pvc_fencing_unfencing.py

```diff
@@ -0,0 +1,632 @@
+
+    @pytest.fixture()
+    def teardown(self, request, nodes):
+        """
+        Make sure all nodes are up again
+        Make sure that all cluster's nodes are in 'Ready' state and if not,
+        change them back to 'Ready' state by restarting the nodes
+
+        """
+        def finalizer():
+            not_ready_nodes = [
+                n for n in node.get_node_objs() if n.ocp.get_resource_status(
+                    n.name
+                ) == constants.NODE_NOT_READY
+            ]
+            logger.info(not_ready_nodes)
+            if not_ready_nodes:
+                nodes.restart_nodes(not_ready_nodes)
+            for node_obj in not_ready_nodes:
+                node.wait_for_nodes_status(
+                    node_names=[node_obj.name], status=constants.NODE_READY, timeout=300
+                )
+
+        request.addfinalizer(finalizer)
+
+    def identify_nodes(self, scenario, num_of_nodes):
+        """
+        Fetches info about the worker nodes and add nodes (if required)
+
+        Args:
+            scenario (str): Scenario of app pods running on OCS or dedicated nodes
+                (eg., 'colocated', 'dedicated')
+            ocs_nodes (list): list of OCS nodes name
+            non_ocs_nodes (list): list of non-OCS nodes name
+            num_of_nodes (int): number of nodes to be selected
+            ignore_check (bool): True for selecting OCS nodes without checking
+                for mon and osd pods, False otherwise
+
+        Returns:
+            worker_nodes_name (list): list of names of worker nodes
+            ocs_nodes_name (list): list of names of ocs nodes
+            non_ocs_nodes_name (list): list of names non-ocs nodes
+        """
+        node_obj = ocp.OCP(kind=constants.NODE)
+        worker_nodes = node_obj.get(selector=constants.WORKER_LABEL).get('items')
+        worker_nodes_name = [node_obj.get('metadata').get('name') for node_obj in worker_nodes]
+        ocs_nodes = node_obj.get(selector=constants.OPERATOR_NODE_LABEL).get('items')
+        ocs_nodes_name = [node_obj.get('metadata').get('name') for node_obj in ocs_nodes]
+        non_ocs_nodes_name = list(set(worker_nodes_name) - set(ocs_nodes_name))
+
+        if 'colocated' in scenario:
+            if len(ocs_nodes_name) < num_of_nodes:
+                if len(worker_nodes_name) >= num_of_nodes:
+                    helpers.label_worker_node(
+                        node_list=non_ocs_nodes_name[0:(num_of_nodes - len(ocs_nodes_name))],
+                        label_key="cluster.ocs.openshift.io/openshift-storage",
+                        label_value="''"
+                    )
+                else:
+                    # Todo: use add_node function when implemented
+                    pytest.skip("Insufficient nodes, Add node support needed")
+
+        else:
+            if len(non_ocs_nodes_name) < num_of_nodes:
+                pytest.skip("Insufficient nodes, Add node support needed")
```

Comment:
maybe would be good to have this check before the setup fixture, to save running time

Reply:
The plan is to do `add node` according to the test requirement.
As currently, we don't have add node functionality in ocs-ci, I'm skipping here.
`test_rwo_pvc_fencing_node_prolonged_network_failure` already have skip marker, I'll also add skip marker to other remaining tests which require more nodes

Reply:
I mean please move this validation to an earlier phase of the test so in case the test needs to be skipped, we won't waste execution time

Reply:
Yes, I'm calling this function identify_and_add_nodes() immediately in the setup [fixture](https://github.com/red-hat-storage/ocs-ci/pull/1416/files#diff-7431b0648296f0708914662200f6923fR33)  at L#33